libyul/AsmAnalysis.cpp

```diff
@@ -499,14 +520,17 @@ size_t AsmAnalyzer::operator()(FunctionCall const& _funCall)
 							m_errorReporter.typeError(
 								7575_error,
 								nativeLocationOf(arg),
-								"Data name \"" + formattedLiteral + "\" cannot be used as an argument of eofcreate/returncontract. " +
-								"An object name is only acceptable."
+								fmt::format(
+									"Data name \"{}\" cannot be used as an argument of eofcreate/returncontract. "
+									"An object name is only acceptable.",
```

Comment:
Re-reading this, the word order is a bit off. I'd rather say "Only an object name is acceptable." Should we fix this here?

Reply:
Yeah, this wording is pretty awkward. Your version is much more comprehensible :)

Reply:
I wonder if we could do something to make messages more consistent and comprehensible in general. They're often pretty bad. And they're the part of the compiler's UI that users see a lot, so it affects how users see us.

In PRs I usually have an urge to just rewrite all of them :) Though normally I try to limit myself to just to those where I can actually point to an actual issue (rather than just vaguely not liking how they sound).

Maybe at least for bigger features it would be good to go over them again when the feature is complete and we're adding docs? Especially putting them all side by side in a comment in such a PR could let us catch weirdness like this before it goes out to the world.

Reply:
Yes I think it is important to streamline them. As you said, they _are_ user-facing and good error messages are super important. They make or break me wanting to continue using a software. 
At some point it comes down to taste of course, I know the urge well :) Putting them into a comment and/or at least having a pass over them before releasing a feature into the wild sounds like a good idea. Besides the functionality itself, I would say it is one of the more important things to get right.

Reply:
Sounds good. Let's try that next time we do a feature that adds new messages.